### PR TITLE
JEC and pseudo top objects

### PR DESCRIPTION
**Pull request by EmyrClement**
# JEC
* Add option to apply a different set of JEC (from the global tag) to the set that is already applied to jets in miniAOD
* Apply selection on jets with the different JEC applied
* Store the jets with the different JEC applied (rather than the old set)

# Pseudo Top and Unfolding
* Add pseudo top/particle level objects to ntuples
* Add generator decay selection.  For ttbar samples, we now store all events that pass a selection criteria or is a semileptonic ttbar event (electron or muon).  This should be enough information for the unfolding.

# Other minor changes:
* Removed hit fit configuration, as it's done in analysis tools
* Put test miniAOD files in hdfs/TopQuarkGroup/run2/miniAOD/

### CODE REVIEW

File: src/BristolNTuple_PFJets.cc [hunk 1]
```diff
@@ -211,6 +223,12 @@ void BristolNTuple_PFJets::produce(edm::Event& iEvent, const edm::EventSetup& iS
 				jecUnc->setJetPt(it->pt()); // the uncertainty is a function of the corrected pt
 			}
 
+			double JEC = 1;
+			if ( readJEC ) {
+				JEC = corrector->correction( it->correctedJet("Uncorrected"), iEvent, iSetup );
```
Comment (kreczko) [inline src/BristolNTuple_PFJets.cc]: This is equivalent to ```TopPairElectronPlusJetsSelectionFilter::getJECForJet()``` and ```TopPairMuonPlusJetsSelectionFilter::getJECForJe()t```

File: src/BristolNTuple_PFJets.cc [hunk 2]
```diff
@@ -343,10 +361,19 @@ void BristolNTuple_PFJets::produce(edm::Event& iEvent, const edm::EventSetup& iS
...
 			//kinematic variables
-			px->push_back(it->px());
-			py->push_back(it->py());
-			pz->push_back(it->pz());
-			energy->push_back(it->energy());
+			if ( readJEC ) {
+				px->push_back(it->correctedJet("Uncorrected").px() * JEC );
+				py->push_back(it->correctedJet("Uncorrected").py() * JEC );
+				pz->push_back(it->correctedJet("Uncorrected").pz() * JEC );
+				energy->push_back(it->correctedJet("Uncorrected").energy() * JEC);
```
Comment (kreczko) [inline src/BristolNTuple_PFJets.cc]: This should be equivalent to ```TopPairElectronPlusJetsSelectionFilter::applyNewJec()``` and ```TopPairMuonPlusJetsSelectionFilter::applyNewJec()```

**Comment by kreczko** [comment]
I would suggest to put
 - ```TopPairElectronPlusJetsSelectionFilter::applyNewJec()``` and ```TopPairMuonPlusJetsSelectionFilter::applyNewJec()```
 - ```TopPairElectronPlusJetsSelectionFilter::getJECForJet()``` and ```TopPairMuonPlusJetsSelectionFilter::getJECForJet()```
into PatUtilities.* to reduce duplicated code.

File: plugins/TopPairElectronPlusJetsSelectionFilter.cc [hunk 3]
```diff
@@ -533,6 +546,24 @@ void TopPairElectronPlusJetsSelectionFilter::cleanedJets() {
...
 }
 
+pat::JetCollection TopPairElectronPlusJetsSelectionFilter::applyNewJec( pat::JetCollection jets, edm::Event& iEvent, const edm::EventSetup& iSetup ) {
+	pat::JetCollection newJets;
+
+	for (unsigned index = 0; index < jets.size(); ++index) {
+		const pat::Jet jet = jets.at(index);
+		float JEC = getJECForJet( jet, iEvent, iSetup);
+		pat::Jet newJet( jet );
+		newJet.setP4( jet.p4() * JEC );
```
Comment (EmyrClement) [inline plugins/TopPairElectronPlusJetsSelectionFilter.cc]: Found a bug here.  It should be:

newJet.setP4( jet.correctedJet("Uncorrected").p4() * JEC );

**Comment by EmyrClement** [comment]
Responded to Luke's comments.

Also fixed bug in application of JEC.  Was not applying JEC on uncorrected jets!

**Pull request merged**